src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -16,21 +19,26 @@
 @Service
 @Transactional
 public class LineService {
-    private LineRepository lineRepository;
-    private StationService stationService;
+    private final LineRepository lineRepository;
+    private final StationService stationService;
 
     public LineService(LineRepository lineRepository, StationService stationService) {
         this.lineRepository = lineRepository;
         this.stationService = stationService;
     }
 
+    @Caching(evict = {
+            @CacheEvict(value = "lines", key = "'lineAllList'")
+    })
```

Comment:
> 1. save를 할때 key가 id로 지정된 캐시들도 지워줘야한다고 생각되서

    @Caching(evict = {
            @CacheEvict(value = "line", key = "#id"),
            @CacheEvict(value = "lines", key = "'lineAllList'")
    })
    public LineResponse saveLine(LineRequest request) {
> 이렇게 @CacheEvict를 넣었는데 아래와 같은 에러가 나옵니다... 아마 @CacheEvict(value = "line", key = "#id") 이걸 넣어서 에러가 나오는거 같은데 아무리 찾아봐도 잘 안나와서요ㅠㅠ 혹시 제가 놓치고 있는게 있을까요??

spEL 문법을 잘못 사용하셔서 에러가 나고 있는 것 같습니다. 
#id를 통해 파라미터에서 id를 찾으려고하는데 존재하지 않기 때문에 에러가 나고 있어요!
아웃사이더님이 번역해주신 spring spEL 레퍼런스 링크 남겨드리겠습니다 😄 
https://blog.outsider.ne.kr/835

> 2. 위와 같이 매개변수에 해당 키값이 없는 캐시를 지워줄려고 할때에는 어떤 방식으로 할 수 있을까요?

아래와 같이 설정하시면 모든 키를 지울 수 있습니다!
다만 노선이 저장될때 외 다른 노선에 대한 캐싱을 제거해야하는 생각을 하신지 궁금합니다 ! 🤔 

```suggestion
    @Caching(evict = {
            @CacheEvict(value = "line", allEntries = true),
            @CacheEvict(value = "lines", key = "'lineAllList'")
    })
```
https://docs.spring.io/spring-framework/docs/5.1.8.RELEASE/spring-framework-reference/integration.html#cache-annotations-evict

Reply:
답변감사합니다!!

아래 질문을 제가 잘못 작성한거같아서요ㅠㅠ 죄송합니다...

노선이아니라 station을 삭제했을때 노선에 연결된 station들도 있는데요 이때에는 노선 캐싱을 지워줘야하는 건가요?

Reply:
노선을 조회할 때 station을 함께 캐싱하고 있다면 지워주는 것이 맞을 것 같습니다 :)